src/compiler/orcv2.jl

```diff
@@ -128,9 +129,15 @@ function move_to_threadsafe(ir)
     buf = convert(MemoryBuffer, ir)
 
     # 2. deserialize and wrap by a ThreadSafeModule
-    return ThreadSafeContext() do ctx
-        mod = parse(LLVM.Module, buf; ctx=context(ctx))
-        ThreadSafeModule(mod; ctx)
+    ts_ctx = ThreadSafeContext()
+    ctx = context(ts_ctx)
+    activate(ctx)
+    try
```

Comment:
@vchuravy @maleadt  is there a nicer way to do this?

Reply:
On Julia < 1.9, JuliaContext() returns a Context() rather than a ThreadSafeContext() so ThreadSafeModule fails. The code snippet here is the same as that used in GPUCompiler for the >= 1.9 case.